### PR TITLE
Adds notice to internals page

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/10873

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- enable rewards
- go to internals page
- make sure that you see a notice

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/resources/brave_components_strings.grd [hunk 1]
```diff
@@ -404,6 +404,7 @@
       <message name="IDS_BRAVE_REWARDS_INTERNALS_CONTRIBUTION_STEP_REWARDS_OFF" desc="">Rewards was turned off</message>
       <message name="IDS_BRAVE_REWARDS_INTERNALS_CONTRIBUTION_STEP_AUTO_CONTRIBUTE_OFF" desc="">Auto-contribute was turned off</message>
       <message name="IDS_BRAVE_REWARDS_INTERNALS_CONTRIBUTION_STEP_RETRY_COUNT" desc="">Stopped retrying</message>
+      <message name="IDS_BRAVE_REWARDS_INTERNALS_MAIN_DISCLAIMER" desc="">All data on this page is sensitive and may include your private keys or wallet IDs. Use abundant caution before sharing.</message>
```
Comment (diracdeltas) [inline components/resources/brave_components_strings.grd:407]: maybe rephrase this as:

`These logs may contain sensitive information. Treat them as you would your wallet private keys. Be careful who you share them with.`

Comment (NejcZdovc) [reply components/resources/brave_components_strings.grd:407]: @diracdeltas we are not talking about just logs here anymore as disclaimer will be shown for all tabs

Comment (NejcZdovc) [reply components/resources/brave_components_strings.grd:407]: or should we just put disclaimer on log tab?

Comment (NejcZdovc) [reply components/resources/brave_components_strings.grd:407]: adjusted text. @diracdeltas this is what I have now

image

**Comment by fmarier** [comment]
> Ideally we should also add a warning at the top of the log file itself, but this doesn't need to block this PR and can be added later.

I filed https://github.com/brave/brave-browser/issues/10888 for this.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 10.15.6 x64` using the following build:

```
Brave | 1.13.42 Chromium: 84.0.4147.89 (Official Build) nightly (64-bit)
-- | --
Revision | 19abfe7bcba9318a0b2a6bc6634a67fc834aa592-refs/branch-heads/4147@{#852}
OS | macOS Version 10.15.6 (Build 19G73)
```

Verification details can be found under https://github.com/brave/internal/issues/713#issuecomment-663351330.
